java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -66,6 +76,18 @@ public Optional<String[]> getColumns() {
     return columns;
   }
 
+  public Optional<String> getFilter() {
+    return filter;
+  }
+
+  public long getStartOffset() {
```

Comment:
well, text files can also split into segments in HADOOP environment.

Reply:
Yes, my point is that if we need to support (optional) file splitting, users should not get surprising result. We may need to implement this for all supported file formats, or throw if the format does not support splitting. It would be fine to leave a TODO comment for other formats for now.

Reply:
TBH, I am not very familiar with dataset, does ScanOptions apply to a single input file or total dataset? My concern is the scope of its use.

Reply:
> We may need to implement this for all supported file formats, or throw if the format does not support splitting.

make sense.  I did bear that in mind, so I am having this new newRangeScan interface to indicate the change, and current behavior wont broke.  I'm not sure I should touch all the format and leave a TODO there? or should I just add it in the interface (also feels a bit weird).  maybe default throw is better.

> does ScanOptions apply to a single input file or total dataset

in my understanding, it applies to a dataset. in that sense it's a bit weird to specify offset and length. let's see if we have better options.

Reply:
I found that the code is parquet-mr is:
1. `ParquetInputSplit`
2. It will generate a `metadataFilter` by range: https://github.com/apache/parquet-mr/blob/d2c3c6d2e761a17b75d56fd356a37a2f754072f7/parquet-hadoop/src/main/java/org/apache/parquet/ParquetReadOptions.java#L39
3. Finally, the filter is: https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java#L1244